### PR TITLE
Add findBySku and updateBySku methods

### PR DESCRIPTION
**Pull request by trisberg**
Adding an `/article/updateBySku` PATCH method so we can update inventory quantities

- only update if the current inventory quantity matches the passed in quantity to update from

- also adding `/article/findBySku` method

- renaming path for delete to `/article/deleteBySku`

- updating UI with the above path

- hiding the custom repository methods from being exposed as part of `/article/search`

### CODE REVIEW

**Comment by trisberg** [comment]
I'll take a crack at adding some tests today and then you can add anything missing tomorrow

**Comment by trisberg** [comment]
Any idea why the CI fails?

```
  [19:40:46] E/launcher - SessionNotCreatedError: session not created: This version of ChromeDriver only supports Chrome version 79
    (Driver info: chromedriver=79.0.3945.36 (3582db32b33893869b8c1339e8f4d9ed1816f143-refs/branch-heads/3945@{#614}),platform=Linux 5.0.0-1025-azure x86_64)
      at Object.checkLegacyResponse (/home/runner/work/inventory-management/inventory-management/gui/node_modules/selenium-webdriver/lib/error.js:546:15)
      at parseHttpResponse (/home/runner/work/inventory-management/inventory-management/gui/node_modules/selenium-webdriver/lib/http.js:509:13)
      at /home/runner/work/inventory-management/inventory-management/gui/node_modules/selenium-webdriver/lib/http.js:441:30
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

File: src/test/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepositoryTest.java [hunk 1]
```diff
@@ -50,6 +50,32 @@ void finds_all_articles() {
...
+
+  @Test
+  @DisplayName("Updates quantity for articles by SKU")
+  void update_article_by_sku() {
+    jdbcHelper.save(article);
+    jdbcHelper.save(otherArticle);
+
+    int count = repository.updateQuantityBySku(article.getSku(), 12, 11);
+
+    Article updatedArticle = new Article("some SKU", "some name", "description", BigDecimal.TEN, null, 11);
```
Comment (fbiville) [inline src/test/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepositoryTest.java]: the constructor argument should be calls to the getters of `article` (except for the quantity), shouldn't they?

File: src/main/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepository.java [hunk 2]
```diff
@@ -10,7 +10,16 @@
 @RepositoryRestResource(path = "article")
 public interface ArticleRepository extends CrudRepository<Article, Long> {
 
+  @Query("SELECT id,sku,name,description,price_in_usd,image_url,quantity FROM article WHERE sku = :sku")
+  Article findBySku(@Param("sku") String sku);
+
+  @RestResource(exported = false)
+  @Modifying
+  @Query("UPDATE article SET quantity = :to WHERE sku = :sku AND quantity = :from")
```
Comment (fbiville) [inline src/main/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepository.java]: now that I think about it, is there a particular reason for requiring the current quantity in the query?

File: src/test/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepositoryTest.java (hunk 1 above)
Comment (trisberg) [reply src/test/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepositoryTest.java]: I though about it but that became a very long statement - I can change that though

File: src/main/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepository.java (hunk 2 above)
Comment (trisberg) [reply src/main/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepository.java]: We only want to change the quantity if it hasn't been updated by someone else while we were processing, so checking that it still is what we think it is seems proper to achieve optimistic concurrency control

**Comment by fbiville** [comment]
Merged as 0218943 and ae2aaec

**Pull request closed without merging**